app/views/questionnaire.py

```diff
@@ -337,28 +337,16 @@ def update_questionnaire_store_with_form_data(questionnaire_store, location, ans
 
             # Dates are comprised of 3 string values
             if isinstance(answer_value, dict):
-                is_day_month_year = 'day' in answer_value and 'month' in answer_value and 'year' in answer_value
-                is_month_year = 'day' not in answer_value and 'year' in answer_value and 'month' in answer_value
-
-                if is_day_month_year and answer_value['day'] and answer_value['month']:
-                    date_str = "{:02d}/{:02d}/{}".format(
-                        int(answer_value['day']),
-                        int(answer_value['month']),
-                        answer_value['year'],
-                    )
-                    answer = Answer(answer_id=answer_id, value=date_str, location=location)
-                elif is_month_year and answer_value['month']:
-                    date_str = "{:02d}/{}".format(int(answer_value['month']), answer_value['year'])
-                    answer = Answer(answer_id=answer_id, value=date_str, location=location)
+                if answer_value_empty(answer_value):
```

Comment:
Maybe it would be better to do this in `update_questionnaire_store_with_form_data` before it's added